dashboard/api.py

```diff
@@ -207,27 +209,29 @@ def _add_run(run, mmtrack_, status):
     run_statuses.remove(run_status)
 
     if run_status.status == CourseRunStatus.NOT_ENROLLED:
-        next_run = course.get_next_run()
+        next_run = course.get_first_unexpired_run()
         if next_run is not None:
             _add_run(next_run, mmtrack, CourseStatus.OFFERED)
     elif run_status.status == CourseRunStatus.NOT_PASSED:
-        next_run = course.get_next_run()
+        _add_run(run_status.course_run, mmtrack, CourseRunStatus.NOT_PASSED)
+        next_run = course.get_first_unexpired_run()
+        if next_run is not None:
+            _add_run(next_run, mmtrack, CourseStatus.OFFERED)
+    elif run_status.status == CourseRunStatus.MISSED_DEADLINE:
+        _add_run(run_status.course_run, mmtrack, CourseRunStatus.MISSED_DEADLINE)
```

Comment:
It looks like this case is uncovered, can you add a test for it?

![screenshot from 2016-10-19 14-41-52](https://cloud.githubusercontent.com/assets/863262/19532552/307a4e60-960a-11e6-8cef-a0e4f2b83923.png)